tmt/steps/execute/__init__.py

```diff
@@ -649,9 +650,31 @@ def load(self) -> None:
         except tmt.utils.FileError:
             self.debug('Test results not found.', level=2)
 
+    def merge_results_rerun(self) -> None:
+        """ Merge new results with old ones for rerun """
+        assert isinstance(self.parent, tmt.base.Plan)  # narrow type
+        old_results: list[Result] = [
+            Result.from_serialized(data) for data in
+            tmt.utils.yaml_to_list(self.read(self.parent.last_run_execute / 'results.yaml'))]
+
+        for result in old_results:
+            # Add old results into new ones and copy log directories
+            self._results.append(result)
+            assert self.workdir is not None  # narrow type
+            assert result.data_path is not None  # narrow type
+            shutil.copytree(
+                self.parent.last_run_execute / result.data_path.parent,
+                self.workdir / result.data_path.parent,
+                dirs_exist_ok=True)
+
+        # Cleanup saved last run data
+        shutil.rmtree(self.parent.last_run_execute, ignore_errors=True)
```

Comment:
Is this needed? What harm does the original run presents to the new one?

Reply:
absolutely none, the question then is what happens on repeated reruns? do we keep all the runs with serial numbers?